### PR TITLE
add banner to dsr

### LINKED ISSUE #50: Add `Banner` Component
Banner is pretty much like `Alert` but is always fixed to either top of the page or at the bottom.

## Props
- `location` - `header` | `footer` 
- `variant` - `static` | `sticky` | `float` | `card`
- `color` - `primary` | `shaded` | `warning` | `danger` | `success` | `info`
- `icon` - optional
- `button` - optional
- `allowDismissal` - bool
- `text` - ReactNode
- `className`
- `id`

## Example

https://tailwindui.com/components/marketing/elements/banners

image

image

image

image

### PR DESCRIPTION
**Pull request by PranavKrishnan007**
Fixes #50

### CODE REVIEW

File: src/components/Banner.tsx [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+      dark: tailwindColors.neutral['400'],
+      contrast: background.negate().toString(),
+      shaded: isDarkTheme ? background.lighten(3).toString() : background.darken(0.6).toString(),
+    };
+
+    return colors[color];
+  }, [theme, color]);
+
+  const textColor = useMemo(
+    () => {
```
Comment (Advaith3600) [inline src/components/Banner.tsx:52]: unwanted new line

File: src/components/Banner.tsx [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+                        'dsr-flex dsr-justify-between',
+                        icon && 'dsr-w-full',
+                      ])}
+                  >
+                      {icon && <Icon icon={icon} size={20} />}
+                      {allowDismissal && (
+                      <button type="button" className="dsr-flex dsr-place-items-center" onClick={onClose}>
+                          <Icon icon={'times'} size={20} />
+                      </button>
+                      )}
```
Comment (Advaith3600) [inline src/components/Banner.tsx:84]: indendation

File: src/components/Banner.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+    return colors[color];
+  }, [theme, color]);
+
+  const textColor = useMemo(
+    () => {
+      return Color(activeColor).isDark() ? '#fff' : '#333';
+    }, [activeColor]);
+
+  const cardRenderer = () => (
```
Comment (Advaith3600) [inline src/components/Banner.tsx:56]: this doesn't have to be a function

File: src/components/Banner.tsx (hunk 2 above)
Comment (Advaith3600) [reply src/components/Banner.tsx:84]: use quotes for "times"

Comment (Advaith3600) [reply src/components/Banner.tsx:84]: use quotes for "times"

File: src/components/Banner.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+                          <Icon icon={'times'} size={20} />
+                      </button>
+                      )}
+                  </div>
+              </div>
+              <p>
+                  {text || 'Banner text'}
+                  {' '}
+                  {learnMore && (
+                      <a href={learnMore.link} className="dsr-whitespace-nowrap  hover:dsr-underline dsr-inline">{learnMore.text}</a>)}
```
Comment (Advaith3600) [inline src/components/Banner.tsx:91]: extra space in between classes

File: src/components/Banner.tsx [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+                variant !== 'full-width' && 'dsr-shadow-lg',
+              ])}
+              style={{ backgroundColor: activeColor, color: textColor }}
+          >
+              <div className="dsr-w-full dsr-flex dsr-gap-4 dsr-place-items-center">
+                  {icon && <Icon icon={icon} size={20} />}
+                  <p>
+                      {text || 'Banner text'}
+                      {' '}
+                      {learnMore && (<a href={learnMore.link} className="dsr-whitespace-nowrap  hover:dsr-underline dsr-inline">{learnMore.text}</a>)}
```
Comment (Advaith3600) [inline src/components/Banner.tsx:123]: extra space

Comment (Advaith3600) [reply src/components/Banner.tsx:123]: no need for the paraenthesis

File: src/components/Banner.tsx [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+          </div>
+      </div>
+  );
+
+  const something = () => {
+    if (variant === 'card') {
+      return cardRenderer();
+    }
+    return bannerRenderer();
+  };
```
Comment (Advaith3600) [inline src/components/Banner.tsx:143]: very useful something 

also make it one line

**Review by Advaith3600 — CHANGES_REQUESTED** [review]

File: src/components/Banner.tsx [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                      <button type="button" className="dsr-flex dsr-place-items-center" onClick={onClose}>
+                          <Icon icon="times" size={20} />
+                      </button>
+                  )}
+              </div>
+          </div>
+      </div>
+  );
+
+  const renderer = () => {if (variant === 'card') return cardRenderer; return bannerRenderer;};
```
Comment (Advaith3600) [inline src/components/Banner.tsx:134]: fix this

**Review by Advaith3600 — CHANGES_REQUESTED** [review]

**Review by aswinshenoy — APPROVED** [review]

**Pull request merged**